### PR TITLE
Add ignore option

### LINKED ISSUE #23: add { ignore: [globs] }
Often the current whitelist globs do the trick, but implementing the blacklist performantly is non-obvious and sometimes not possible.

I'm leaving this as a place-holder, I intend to implement this sometime. But if someone beats me to it, I will not be offended.

---

My thoughts are as follows:

```
walkSync(root, { ignore: [array, of, globs/strings] });
```

Similar to, https://github.com/joliss/node-walk-sync/blob/master/index.js#L53-L55 we would just prune a dir early. And instead of `mayContain` I believe we must use an explicit `match`, which `MatcherCollection` [already supports](https://github.com/stefanpenner/matcher-collection/blob/master/index.js#L13-L21)

Then finally before appending to the output paths, a full check should occur.

All of this should be pay as you go, if people don't use it we should avoid those branches.

### PR DESCRIPTION
**Pull request by alexlafroscia**
Closes #23

### CODE REVIEW

**Comment by alexlafroscia** [comment]
I swapped out `node-glob` for `walk-sync` with these changes in that PR that I'm working on, and it's quite a bit faster! The `ember-cli-eslint` install time (that does the search for `.jshintrc` files) runs in `3.35s` instead of `5.75s` so I'm happy with that.

File: index.js [hunk 1]
```diff
@@ -84,14 +89,15 @@ function _walkSync(baseDir, options, _relativePath) {
 
   for (var i=0; i<sortedEntries.length; ++i) {
     var entry = sortedEntries[i];
+    var isIgnored = (ignoreMatcher && ignoreMatcher.match(entry.relativePath));
```
Comment (stefanpenner) [inline index.js]: Should we simply do `if (isIgnored) { continue; }`

Comment (alexlafroscia) [inline index.js]: Absolutely, makes sense.

File: index.js [hunk 2]
```diff
@@ -84,14 +89,17 @@ function _walkSync(baseDir, options, _relativePath) {
 
   for (var i=0; i<sortedEntries.length; ++i) {
     var entry = sortedEntries[i];
+    if (ignoreMatcher && ignoreMatcher.match(entry.relativePath)) {
```
Comment (stefanpenner) [inline index.js]: Although correct, we may want to do this check between line 67 and 68. This prevents us from doing costly the costly `getStat`.

Comment (alexlafroscia) [inline index.js]: Oh, I see... Prevent it from ever being added to `entries` instead, right?  Makes sense to me.

**Comment by stefanpenner** [comment]
@alexlafroscia perfect, thank you!

**Comment by stefanpenner** [comment]
released as v0.3.0

**Comment by joliss** [comment]
Hm, what do you want to happen when you pass both `globs` and `ignore`? Reading the code, it would seem that right now, weird things happen.

I think we should either make them work together, or document that they're exclusive and throw an error if both are passed.

**Comment by joliss** [comment]
Hm. Ignore me. Sorry!

**Pull request merged**